File: frontend/src/types/project.ts
/**
 * Project, Quote, and Account type definitions
 */
import type { BaseEntity } from './common';

export interface Quote extends BaseEntity {
  job_id: string;
  name: string;
  customer_id?: number;
  engineer_id?: number;
  salesman_id?: number;
  date: string;
  sell_price: number;
  status: string;
  customer_name?: string;
  engineer_name?: string;
  salesman_name?: string;
}

export interface QuoteCreate {
  job_id: string;
  name: string;
  customer_id?: number;
  engineer_id?: number;
  salesman_id?: number;
  date: string;
  sell_price: number;
  status: string;
}

export interface Project extends BaseEntity {
  project_id: string;
  name: string;
  customer_id?: number;
  engineer_id?: number;
  end_user?: string;
  date: string;
  salesman_id?: number;
  status: string;
  customer_name?: string;
  engineer_name?: string;
  salesman_name?: string;
}

export interface ProjectCreate {
  project_id: string;
  name: string;
  customer_id?: number;
  engineer_id?: number;
  end_user?: string;
  date: string;
  salesman_id?: number;
  status: string;
}

export interface CustomerAccount extends BaseEntity {
  invoice_number: string;
  date: string;
  project_id?: number;
  customer_id?: number;
  name: string;
  amount: number;
  outstanding: number;
  reminder_date?: string;
  comments?: string;
  project_name?: string;
  customer_name?: string;
}

export interface CustomerAccountCreate {
  invoice_number: string;
  date: string;
  project_id?: number;
  customer_id?: number;
  name: string;
  amount: number;
  outstanding: number;
  reminder_date?: string;
  comments?: string;
}

File: backend/models/project.py
from pydantic import BaseModel
from typing import Optional

class Quote(BaseModel):
    id: Optional[int] = None
    job_id: str
    name: str
    customer_id: Optional[int] = None
    engineer_id: Optional[int] = None
    salesman_id: Optional[int] = None
    date: str
    sell_price: float = 0.00
    status: str = "Draft"

class QuoteCreate(BaseModel):
    job_id: str
    name: str
    customer_id: Optional[int] = None
    engineer_id: Optional[int] = None
    salesman_id: Optional[int] = None
    date: str
    sell_price: float = 0.00
    status: str = "Draft"

class Project(BaseModel):
    id: Optional[int] = None
    project_id: str
    name: str
    customer_id: Optional[int] = None
    engineer_id: Optional[int] = None
    end_user: Optional[str] = None
    date: str
    salesman_id: Optional[int] = None
    status: str = "Active"

class ProjectCreate(BaseModel):
    project_id: str
    name: str
    customer_id: Optional[int] = None
    engineer_id: Optional[int] = None
    end_user: Optional[str] = None
    date: str
    salesman_id: Optional[int] = None
    status: str = "Active"

class CustomerAccount(BaseModel):
    id: Optional[int] = None
    invoice_number: str
    date: str
    project_id: Optional[int] = None
    customer_id: Optional[int] = None
    name: str
    amount: float = 0.00
    outstanding: float = 0.00
    reminder_date: Optional[str] = None
    comments: Optional[str] = None

class CustomerAccountCreate(BaseModel):
    invoice_number: str
    date: str
    project_id: Optional[int] = None
    customer_id: Optional[int] = None
    name: str
    amount: float = 0.00
    outstanding: float = 0.00
    reminder_date: Optional[str] = None
    comments: Optional[str] = None
